public/app/features/alerting/unified/hooks/useFilteredRules.ts

```diff
@@ -0,0 +1,60 @@
+import { useMemo } from 'react';
+
+import { CombinedRuleGroup, CombinedRuleNamespace } from 'app/types/unified-alerting';
+import { isCloudRulesSource } from '../utils/datasource';
+import { isAlertingRule } from '../utils/rules';
+import { getFiltersFromUrlParams } from '../utils/misc';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+
+export const useFilteredRules = (namespaces: CombinedRuleNamespace[]) => {
+  const [queryParams] = useQueryParams();
+  const filters = getFiltersFromUrlParams(queryParams);
+
+  return useMemo(() => {
+    const filteredNamespaces = namespaces
+      .filter(({ rulesSource }) =>
+        filters.dataSource && isCloudRulesSource(rulesSource) ? rulesSource.name === filters.dataSource : true
```

Comment:
some more logic will be needed here once my PR is merged; grafana-managed rules can reference multiple data sources. Maybe add a @TODO? I can take care of it later